cdap-api/src/main/java/co/cask/cdap/api/artifact/ArtifactVersion.java

```diff
@@ -30,11 +30,12 @@
  * parse version string in format of [major].[minor].[fix](-|.)[suffix]
  * }
  * </pre>
+ * Note that the suffix can not have numerical values in it.
  */
 @Beta
 public final class ArtifactVersion implements Comparable<ArtifactVersion> {
 
-  private static final Pattern PATTERN = Pattern.compile("(\\d+)(?:\\.(\\d+))?(?:\\.(\\d+))?(?:[.\\-](.*))?");
+  private static final Pattern PATTERN = Pattern.compile("(\\d+)(?:\\.(\\d+))?(?:\\.(\\d+))?(?:[.\\-]([^\\d]*))?$");
```

Comment:
Even in this case:
"cdap-etl-3.5-batch-1.0-SNAPSHOT"
"3.5-batch-1.0-SNAPSHOT" is being matched as the full version string.
3 is major version, 5 is minor version, and remainder is suffix.

You brought up a good point that suffix needs to allow numbers (because it might have timestamp in it), though.